app/src/main/java/com/stevesoltys/seedvault/transport/backup/PackageService.kt

```diff
@@ -29,29 +32,39 @@ private const val LOG_MAX_PACKAGES = 100
 internal class PackageService(
     private val context: Context,
     private val backupManager: IBackupManager,
-) {
+) : KoinComponent {
 
     private val packageManager: PackageManager = context.packageManager
     private val myUserId = UserHandle.myUserId()
+    private val backupCoordinator by inject<BackupCoordinator>()
```

Comment:
Instead of injecting the `BackupCoordinator` here in a cyclic dependency (which will blow up once we consistently use an injection framework such as Hilt), could we move the `isAppEligibleForBackup` code into another place where it then gets used by `PackageService` and `BackupCoordinator`? Maybe `SettingsManager` would work?

Reply:
Okay, a few issues, based on a work in progress:
1. `SettingsManager` does not have access to the current storage plugin, which `isAppEligibleForBackup` needs in order to exclude the storage plugin's package name. I've decided to try passing the `plugin` to it. I'm not sure if that's acceptable.
2. Moving `isAppEligibleForBackup` to `SettingsManager` also means that the test needs to be updated, likely moved out of `BackupCoordinatorTest` and into `SettingsManagerTest`, which does not currently exist. I've started creating it.
3. The existing test relies on `packageInfo`, which is defined in `TransportTest`. It feels possibly odd to have `SettingsManagerTest` test inherit from `TransportTest`, but I can either try that or I can re-define some test package info in `SettingsManagerTest`. The former is nice to prevent code duplication, but maybe there's some better way.

Separately from all of this, I'd like to rename `eligiblePackages` and `isAppEligibleForBackup` to `requestedPackages` and `isAppRequestedForBackup`, respectively, to better reflect their new roles and make it a little less confusing.

Reply:
...Done, but the patch is much bigger now, unfortunately. This wasn't easy. I don't mind undoing it if needed. Learned a little more about mockk along the way.